arkouda/strings.py

```diff
@@ -55,7 +55,13 @@ class Strings:
     -----
     Strings is composed of two pdarrays: (1) offsets, which contains the
     starting indices for each string and (2) bytes, which contains the 
-    raw bytes of all strings, delimited by nulls.    
+    raw bytes of all strings, delimited by nulls.
+    
+    Warnings
+    --------
+    As causing some errors, you should not implement the __del__() method.
+    Instead the self.entry attribute is a pdarray which gets cleaned up
+    automatically through its __del__() method.
```

Comment:
I think speaking a bit more to the reason we don't want to implement a `__del__` method would be useful for any devs who are wondering why it's missing. A good potential rewording for this is something like:

```python
"""
NOTE:
  The Strings.__del__() method should NOT be implemented. Python will invoke the __del__() of any components by default.
  Overriding this default behavior with an explicitly specified Strings.__del__() method may introduce errors in the event that additional components are added to Strings and the method is not updated.
  By allowing Python's garbage collecting to handle this automatically, we avoid extra maintenance
"""
```

I also think moving this message to be right before the `__iter__` method (line 178 of strings.py) would be most useful (since this is where I would likely try to put an `__del__` method). Though I could be convinced otherwise if others have differing opinions

https://github.com/Bears-R-Us/arkouda/blob/d0fda3917835e46f3a6abe080bd8a1cc3fa377d2/arkouda/strings.py#L178-L180

Reply:
@Pl4tt are you going to make the change so we can get your PR accepted and merged?

Reply:
sure, I'll do it straight away :) Thanks for your feedback!